TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst

```diff
@@ -226,14 +236,24 @@ The second method, ``createFromEnvironment()``, take the environment provided by
 functions into account. It creates a ``Request`` instance which reflects the current HTTP request received from the web
 server. This method is best used if you need a ``Request`` object with all properties set according to the current
 server environment and incoming HTTP request.
+Note though, that you should not expect this ``Request`` to match the current ``Request``, since the latter will still
+have been affected by some ``HTTP Components``. If you need the **current** Request, get it from the ``RequestHandler`` instead.
 
 Creating an ActionRequest
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
 In order to dispatch a request to a controller, you need an ``ActionRequest``.
 Such a request is always bound to an ``Http\Request``::
 
-	$httpRequest = Request::createFromEnvironment();
+	/**
+	 * @var HttpRequestHandlerInterface
+	 * @Flow\Inject
+	 */
+	protected $requestHandler;
```

Comment:
Thanks for the hint! 
But couldn't we check for that with `if ($requestHandler instanceof HttpRequestHandlerInterface) {` I've seen other places where we do that.

About `HttpRequestHandlerInterface`:
Now I get ```Uncaught Exception in Flow
Argument 1 passed to TYPO3\Flow\Http\RequestHandler::__construct() must be an instance of TYPO3\Flow\Core\Bootstrap, none given, called in /var/www/kaprunerhof.at/kaprunerhof.getan.at/Packages/Framework/TYPO3.Flow/Classes/TYPO3/Flow/Object/ObjectManager.php on line 530```